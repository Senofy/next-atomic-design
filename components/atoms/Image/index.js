import PropTypes from 'prop-types'
import styled from 'styled-components'
import { font, palette } from 'styled-theme'
import Image from 'next/image'

export const Img = styled(Image)`
    
`

Img.propTypes = {
    reverse: PropTypes.bool,
}

Img.defaultProps = {
    palette: 'grayscale',
}