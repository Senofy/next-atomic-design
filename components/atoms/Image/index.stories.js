import { storiesOf } from '@storybook/react'
import { Img } from './'

storiesOf('Atom', module)
    .add('default', () => (
        <Img>Test</Img>
    ))
