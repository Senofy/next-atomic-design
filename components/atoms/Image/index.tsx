import styled from 'styled-components'
import {
	borderRadius,
	BorderRadiusProps,
	color,
	ColorProps,
	height,
	HeightProps,
	maxWidth,
	MaxWidthProps,
	width,
	WidthProps,
} from 'styled-system'

export interface ImageProps extends WidthProps, MaxWidthProps, HeightProps, BorderRadiusProps, ColorProps {
	src: string
	objectFit?: string
	transform?: string
}

export const Img = styled.img<ImageProps>`
	${borderRadius};
	${color};
	${width};
	${maxWidth};
	${height};
	transform: ${({ transform }) => transform};
	object-fit: ${({ objectFit }) => objectFit};
`
