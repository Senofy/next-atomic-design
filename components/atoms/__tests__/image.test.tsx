import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Img } from '../Image'

afterEach(cleanup)

const defaultProps = {
	src: '/img/sample/cover-image.png',
	height: 150,
	width: 150,
	alt: 'Cover image',
}

test('<Img> should take a snapshot', () => {
	const { asFragment } = render(<Img {...defaultProps} />)
	expect(asFragment()).toMatchSnapshot()
})

test('<Img> renders children when passed in', () => {
	const { getByAltText } = render(<Img {...defaultProps} />)
	expect(getByAltText(/cover image/i)).toBeInTheDocument()
})
