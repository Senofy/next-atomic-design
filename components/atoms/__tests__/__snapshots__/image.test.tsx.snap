// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
  >
    <div
      style="box-sizing: border-box; display: block; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        role="presentation"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTUwIiBoZWlnaHQ9IjE1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
        style="max-width: 100%; display: block;"
      />
    </div>
    <img
      alt="Cover image"
      class="Image__Img-k5bfp2-0 jMwuzy"
      decoding="async"
      src="/_next/image?url=%2Fimg%2Fsample%2Fcover-image.png&w=768&q=75"
      srcset="/_next/image?url=%2Fimg%2Fsample%2Fcover-image.png&w=320&q=75 1x, /_next/image?url=%2Fimg%2Fsample%2Fcover-image.png&w=768&q=75 2x"
      style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
  </div>
</DocumentFragment>
`;
